src/api/api.js

```diff
@@ -0,0 +1,159 @@
+import { useEffect, useState } from 'react'
+
+const URL = (idInstance, apiTokenInstance, action) =>
+  `https://api.green-api.com/waInstance${idInstance}/${action}/${apiTokenInstance}`
+
+const auth = (id, token, setID, setToken) => {
```

Comment:
судя по тому, что внутри используешь реакт хуки, у тебя на выходе получлся реакт хук, как минимум он должен начинаться с use тогда
```suggestion
const useAuth = (id, token, setID, setToken) => {
```


Но вообще, получилась все в кучу, часть состояния внутри функции, часть снаружи передается (setToken, setID) + работа с localstorage.
Работа с состоянием - это логика компонентов, а здесь апи сервис, здесь должна быть только логика взаимодействия с апи: сделать запрос, получить ответ, как то данные ответа обработать, в нужный формат привести и отдать назад в компонент.

В таком виде оставлять не стоит